### PR TITLE
Data Type Documentation Fixes

### PR DESCRIPTION
**Pull request by ehsteve**
You can see a preview of the docs here http://ehsteve-sunpy.readthedocs.org/en/docfix_06/

### CODE REVIEW

**Comment by ehsteve** [comment]
@Alex-Ian-Hamilton ready for you.

**Comment by ehsteve** [comment]
@Cadair @ayshih @wafels this is ready for final review! and then on to packaging for release! @Cadair where is the wcsaxes fix that we were promised earlier in the week (not yours)?

File: doc/source/code_ref/map.rst [hunk 1]
```diff
@@ -72,11 +76,16 @@ parameters are met.
...
+
+Instrument Map Classes
+----------------------
+
 .. automodapi:: sunpy.map.sources
 
 
-Writing a new Map Class
------------------------
+Writing a new Insteument Map Class
```
Comment (wafels) [inline doc/source/code_ref/map.rst]: Instrument, not "Insteument"

File: doc/source/guide/data_types/lightcurve.rst [hunk 2]
```diff
@@ -116,7 +60,124 @@ the 'index' keyword, as is shown below.
...
+more complicated figure. For a bit more information about this and some
+examples see :ref:`plotting`. Here is one a more complicated example
+which makes use of this methodology.
+
+.. plot::
+    :include-source:
+
+    from sunpy.lightcurve import GOESLightCurve
+    from sunpy.time import TimeRange
+    tr = TimeRange('2013/07/21', '2013/07/22')
```
Comment (Cadair) [inline doc/source/guide/data_types/lightcurve.rst]: There should be blank lines between the imports and the code in these examples to make them easier to read. Especially if the user downloads the file.

I am not going to block merging the PR over it, but we should try and do it in the future, and correct them as we find them.

File: doc/source/guide/data_types/maps.rst [hunk 3]
```diff
@@ -9,7 +9,7 @@ the :doc:`/code_ref/map` for a more thorough look at SunPy maps.
 There you can see what instruments are currently supported or you can access the
 code reference for each instrument-specific map subclass.
 
-1. Creating maps
+Creating maps
 ----------------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: over long title line.

File: doc/source/guide/data_types/maps.rst [hunk 4]
```diff
@@ -38,7 +38,7 @@ default FITS keywords and return a `~sunpy.map.GenericMap` but results
 may vary. SunPy can also create maps from the jpg2000 files from
 `helioviewer.org <http://helioviewer.org/>`_.
 
-2. Creating Custom Maps
+Creating Custom Maps
 -----------------------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 5]
```diff
@@ -52,7 +52,7 @@ values as assumed. Here is a simple example::
 
 The keys in the header follows the `FITS standard <http://fits.gsfc.nasa.gov/fits_dictionary.html>`_.
 
-3. Inspecting maps
+Inspecting maps
 ------------------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 6]
```diff
@@ -84,11 +87,11 @@ The meta data for the map is accessed by ::
 This references the meta data dictionary with the header information as read
 from the source file.
 
-4. Getting at the data
+Getting at the data
 ----------------------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 7]
```diff
@@ -126,7 +134,7 @@ but accessing the data array directly. For example::
 
     my_map.data.std()
 
-5. Plotting
+Plotting
 -----------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 8]
```diff
@@ -151,7 +159,7 @@ examples see :ref:`plotting`.
    is not met, when the map is plotted a warning will be issued. You can create
    an oriented map by using `~sunpy.map.GenericMap.rotate()` before you plot the Map.
 
-7. Plotting Keywords
+Plotting Keywords
 ********************
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 9]
```diff
@@ -191,7 +199,7 @@ dictionary. In the following example we change the title of the plot by changing
     plt.show()
 
 
-8. Colormaps and Normalization
+Colormaps and Normalization
 ******************************
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 10]
```diff
@@ -206,7 +214,7 @@ The Map object chooses the appropriate colormap for you when it is created as
 long as it recognizes the instrument. To see what colormaps are available::
 
     import sunpy.cm
-    sunpy.cm.cmlist.keys()
+    sunpy.cm.cm.cmlist.keys()
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: this should be fixed in the code not in the docs.

File: doc/source/guide/data_types/maps.rst [hunk 11]
```diff
@@ -282,7 +290,7 @@ Note how the color in the colorbar does not change since these two maps share
 the same colormap while the data values associated with each color do because
 the normalization is different.
 
-8. Masking and Clipping Data
+Masking and Clipping Data
 ----------------------------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 12]
```diff
@@ -443,11 +451,11 @@ addition of an associated boolean array which holds the mask.
     plt.colorbar(extend='both')
 
 
-6. Composite Maps and Overlaying Maps
+Composite Maps and Overlaying Maps
 -------------------------------------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 13]
```diff
@@ -477,7 +485,7 @@ and then plots the result.
 
 This is not a particularly pretty plot but it shows what SunPy can do!
 
-7. Working with your map
+Working with your map
 ------------------------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 14]
```diff
@@ -531,7 +539,7 @@ The meta data from each map can be obtained using::
 This returns a list of map meta objects that have the same order as
 the maps in the `~sunpy.map.MapCube`.
 
-9. Coalignment of Mapcubes
+Coalignment of Mapcubes
 --------------------------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/data_types/maps.rst [hunk 15]
```diff
@@ -572,7 +580,7 @@ Shifts calculated using calculate_match_template_shift can be passed directly
 to the coalignment function.
 
 
-10. Compensating for solar rotation in Mapcubes
+Compensating for solar rotation in Mapcubes
 -----------------------------------------------
```
Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: too long title line.

File: doc/source/guide/installation/linux.rst [hunk 16]
```diff
@@ -5,11 +5,16 @@ Linux
 Overview
 --------
 
+.. warning::
```
Comment (Cadair) [inline doc/source/guide/installation/linux.rst]: have we put one of these in the windows section?

I would also maybe reword this a little bit, especially for Linux where there are other good package management options.

Maybe something like:

```
We highly recommend that linux users install the Anaconda python distribution which is described in the instructions here :ref:`anaconda_install`.
```

File: doc/source/guide/tour.rst [hunk 17]
```diff
@@ -11,43 +11,31 @@ thorough look at the functionality available.
...
-    sunpy.data.download_sample_data()
-
-After running this you can then import the sample data files shortcuts which
-are used below (e.g. sunpy.data.sample) by simply importing the module like so::
-
-    import sunpy.data.sample
-
-If the sample files are not available for some reason that you will get an error
-on import.
+download. If you are already done so please follow the instruction to :ref:`sample-data`.
```
Comment (Cadair) [inline doc/source/guide/tour.rst]: This sentence needs rewording.

File: doc/source/guide/tour.rst [hunk 18]
```diff
@@ -11,43 +11,31 @@ thorough look at the functionality available.
...
 **Creating a Map**
 
-SunPy supports many different data products from various sources 'out of the box' we
+SunPy supports many different data products from various sources 'out of the box'. We
 shall use SDO's AIA instrument as an example in this tutorial. The general way to create
-a map from one of the supported data products is with the `Map()` class from the `map` submodule.
-
-`Map()` takes either a filename, a list of filenames or a data array and header pair. We can test map with:
+a map from one of the supported data products is with the `~sunpy.map.map()` function from the `~sunpy.map` submodule.
+`~sunpy.map.map()` takes either a filename, a list of filenames or a data array and header. We can test map with:
```
Comment (Cadair) [inline doc/source/guide/tour.rst]: missing capital on second `Map`

File: doc/source/guide/tour.rst [hunk 19]
```diff
@@ -11,43 +11,31 @@ thorough look at the functionality available.
...
 
 .. plot::
     :include-source:
-    
+
     import sunpy.data.sample
     import sunpy.map
     aia = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
```
Comment (Cadair) [inline doc/source/guide/tour.rst]: line break again. The ones with include-source are even more important.

File: doc/source/guide/tour.rst [hunk 20]
```diff
@@ -94,11 +82,13 @@ Below is the example built into sunpy.
...
+and the :ref:`spectra_code_ref`.
 
 Plotting
 --------
 
-SunPy uses a matplotlib like interface to it's plotting so more complex plots can be built by combining
+SunPy uses a matplotlib-like interface for it's plotting so more complex plots can be built by combining
 SunPy with matplotlib.
 
 .. plot::
```
Comment (Cadair) [inline doc/source/guide/tour.rst]: this example really could use some whitespace!

**Comment by Cadair** [comment]
@Alex-Ian-Hamilton I think it was your title modifications where you did not truncate the underlines. Just for future reference, it's good practise to do so.

File: doc/source/guide/data_types/maps.rst (hunk 10 above)
Comment (ehsteve) [inline doc/source/guide/data_types/maps.rst]: Do you want to fix this for 0.6? Otherwise I leaves it!

File: doc/source/guide/installation/linux.rst (hunk 16 above)
Comment (ehsteve) [inline doc/source/guide/installation/linux.rst]: What happened to the windows section (win.rst). Looks like it is gone in master? Was this on purpose?

Comment (Cadair) [inline doc/source/guide/installation/linux.rst]: oh, maybe I just deleted it completely in favour of anaconda. This seems like something I might have done...

File: doc/source/guide/installation/advanced.rst [hunk 21]
```diff
@@ -159,8 +156,7 @@ correctly is to use the :func:`sunpy.self_test()` function::
     import sunpy
     sunpy.self_test(online=False)
 
-Which will run all the SunPy tests that don't require an active internet
-connection (and can therefore be quite slow).
+Which will run many of the SunPy tests.
```
Comment (wafels) [inline doc/source/guide/installation/advanced.rst]: lower case "which"

File: doc/source/guide/installation/index.rst [hunk 22]
```diff
@@ -17,46 +17,45 @@ Python / Scientific Python environment then you can skip to the
...
-To configure conda for sunpy downloads run::
+To install SunPy launch a system command prompt or the
+'Anaconda Command Prompt' (under Windows). First configure conda
+for sunpy downloads::
 
     conda config --add channels sunpy --add channels astropy
 
-To install SunPy run::
- 
+theh install SunPy::
```
Comment (wafels) [inline doc/source/guide/installation/index.rst]: "theh" ???

File: sunpy/map/mapbase.py [hunk 23]
```diff
@@ -625,12 +647,12 @@ def resample(self, dimensions, method='linear'):
...
         Returns
         -------
-        out : Map
+        out : `~sunpy.map.GenericMap` or subclass
             A new Map which has been resampled to the desired dimensions.
 
         References
         ----------
-        | http://www.scipy.org/Cookbook/Rebinning (Original source, 2011/11/19)
+        * `Rebinninb <http://www.scipy.org/Cookbook/Rebinning>`_ (Original source, 2011/11/19)
```
Comment (ayshih) [inline sunpy/map/mapbase.py]: typo?

File: sunpy/net/vso/vso.py [hunk 24]
```diff
@@ -806,6 +807,19 @@ def unknown_method(self, response):
...
+        Parameters
+        ----------
+            choices : ?
+
+            response : ?
+
+        Returns
+        -------
+            ? : ?
+        """
```
Comment (ayshih) [inline sunpy/net/vso/vso.py]: Um, can we have something better than question marks here?  Even a phrase like "not yet documented" would be better than "?".

File: sunpy/net/vso/vso.py [hunk 25]
```diff
@@ -828,6 +842,19 @@ def multiple_choices(self, choices, response):
...
+        Parameters
+        ----------
+            info : ?
+
+            field : ?
+
+        Returns
+        -------
+            ? : ?
+        """
```
Comment (ayshih) [inline sunpy/net/vso/vso.py]: Same as above

File: sunpy/roi/chaincode.py [hunk 26]
```diff
@@ -60,44 +60,75 @@ def __init__(self, origin, chaincode, xdelta=1, ydelta=1):
...
+
+        Parameters
+        ----------
+        end : ?
+
+        Returns
+        -------
+        ?
+
+        """
```
Comment (ayshih) [inline sunpy/roi/chaincode.py]: Same thing about question marks

File: sunpy/roi/chaincode.py [hunk 27]
```diff
@@ -60,44 +60,75 @@ def __init__(self, origin, chaincode, xdelta=1, ydelta=1):
...
+        Parameters
+        ----------
+        coordinates : ?
+        index : ?
+
+        Returns
+        -------
+        ?
+
+        """
```
Comment (ayshih) [inline sunpy/roi/chaincode.py]: Same thing about question marks

File: sunpy/time/time.py [hunk 28]
```diff
@@ -216,7 +216,7 @@ def is_time(time_string, time_format=''):
     time_string : [ int, float, time_string, datetime ]
         Date to parse which can be either time_string, int, datetime object.
     time_format : [ basestring, utime, datetime ]
-	Specifies the format user has provided the time_string in.
+    Specifies the format user has provided the time_string in.
```
Comment (ayshih) [inline sunpy/time/time.py]: indention?

File: doc/source/guide/tour.rst [hunk 29]
```diff
@@ -94,11 +82,13 @@ Below is the example built into sunpy.
...
     image.peek()
 
+For more information about spectra, check out the :doc:`spectra guide <data_types/spectra>`
+and the :ref:`spectra_code_ref`.
 
 Plotting
 --------
 
-SunPy uses a matplotlib like interface to it's plotting so more complex plots can be built by combining
+SunPy uses a matplotlib-like interface for it's plotting so more complex plots can be built by combining
```
Comment (ayshih) [inline doc/source/guide/tour.rst]: "its"

File: sunpy/roi/chaincode.py (hunk 27 above)
Comment (wafels) [inline sunpy/roi/chaincode.py]: That was me.  Intended as a placeholder, since in some cases it wasn't
completely clear to me what those functions were meant to do, or what the
parameters were.

I will replace all the ? with "not yet documented" and a todo directive
through a PR to the branch.

**Comment by ayshih** [comment]
I see that some of the docstrings that are largely just question marks have todo blocks as well.  How about extending that to all such question-mark docstrings?

**Comment by ayshih** [comment]
:+1:

**Comment by Cadair** [comment]
I would rather have no docs than placeholders. To dos are a good idea, how do you "use" them though? Would issues not be better?

**Comment by Cadair** [comment]
:+1:

**Comment by Cadair** [comment]
NOPE!

https://travis-ci.org/sunpy/sunpy/jobs/72862439#L1445

You broke it!

**Comment by ehsteve** [comment]
@Cadair TRAVIS is now breaking for something totally unrelated I think.

**Comment by Cadair** [comment]
I can confirm that the ci fail is now present in master :(

**Comment by ehsteve** [comment]
huzzah!
